metagraph/src/annotation/binary_matrix/base/binary_matrix.hpp

```diff
@@ -32,9 +35,13 @@ class BinaryMatrix {
     virtual SetBitPositions get_row(Row row) const = 0;
     virtual std::vector<SetBitPositions> get_rows(const std::vector<Row> &rows) const;
     virtual std::vector<Row> get_column(Column column) const = 0;
+
     // get all selected rows appended with -1 and concatenated
     virtual std::vector<Column> slice_rows(const std::vector<Row> &rows) const;
 
+    virtual void slice_columns(const std::vector<Column> &columns,
+                               const ColumnCallback &callback) const;
```

Comment:
why not just `std::function<void(bitmap&&)> ColumnCallback`?

Reply:
The columns may not be called in the same order as `columns`, so I set up the callback to have a pair with the column index.